pkg/prowgen/prowgen_konflux.go

```diff
@@ -342,10 +347,97 @@ func GenerateKonfluxServerlessOperator(ctx context.Context, openshiftRelease Rep
+				return fmt.Sprintf("&& ("+
+					" files.all.exists(x, x.matches('^olm-catalog/serverless-operator-index/v%s/')) ||"+
+					" files.all.exists(x, x.matches('^.tekton/'))"+
+					" )", v)
+			},
+			AdditionalComponentConfigs: []konfluxgen.TemplateConfig{
+				{
+					ReleaseBuildConfiguration: cioperatorapi.ReleaseBuildConfiguration{
+						Metadata: cioperatorapi.Metadata{
+							Org:    r.Org,
+							Repo:   r.Repo,
+							Branch: branch,
+						},
+						Images: []cioperatorapi.ProjectDirectoryImageBuildStepConfiguration{
+							{
+								To: cioperatorapi.PipelineImageStreamTagReference(fmt.Sprintf("serverless-index-%s-fbc-%s", release, v)),
+								ProjectDirectoryImageBuildInputs: cioperatorapi.ProjectDirectoryImageBuildInputs{
+									DockerfilePath: "Dockerfile",
+									ContextDir:     fmt.Sprintf("./olm-catalog/serverless-operator-index/v%s", v),
+								},
+							},
+						},
+					},
+				},
+			},
+			ComponentNameFunc: func(cfg cioperatorapi.ReleaseBuildConfiguration, ib cioperatorapi.ProjectDirectoryImageBuildStepConfiguration) string {
+				return string(ib.To)
+			},
+			ResourcesOutputPathSkipRemove: true,
+			PipelinesOutputPathSkipRemove: true,
+			IsHermetic: func(_ cioperatorapi.ReleaseBuildConfiguration, _ cioperatorapi.ProjectDirectoryImageBuildStepConfiguration) bool {
+				return true
+			},
+			Tags: []string{soMetadata.Project.Version},
+		}
+
+		if err := konfluxgen.Generate(c); err != nil {
+			return fmt.Errorf("failed to generate Konflux FBC configurations for %s (%s): %w", r.RepositoryDirectory(), branch, err)
+		}
+	}
+
+	return nil
+}
+
+func getOPMImage(v string) (string, error) {
+	parts := strings.SplitN(v, ".", 2)
+	if len(parts) != 2 {
+		return "", fmt.Errorf("invalid OCP version: %s", v)
+	}
+
+	minor, err := strconv.Atoi(parts[1])
+	if err != nil {
+		return "", fmt.Errorf("could not convert OCP minor to int (%q): %w", v, err)
+	}
+
+	if minor <= 14 {
+		return fmt.Sprintf("registry.redhat.io/openshift4/ose-operator-registry:v4.%d", minor), nil
+	} else {
+		// use RHEL9 variant for OCP version >= 4.15
+		return fmt.Sprintf("registry.redhat.io/openshift4/ose-operator-registry-rhel9:v4.%d", minor), nil
+	}
+}
+
 func serverlessBundleNudge(downstreamVersion string) string {
 	return konfluxgen.Truncate(konfluxgen.Sanitize(fmt.Sprintf("%s-%s", "serverless-bundle", downstreamVersion)))
 }
 
-func serverlessIndexNudge(downstreamVersion string) string {
-	return konfluxgen.Truncate(konfluxgen.Sanitize(fmt.Sprintf("%s-%s", "serverless-index", downstreamVersion)))
+func serverlessIndexNudges(downstreamVersion string, ocpVersions []string) []string {
+	indexes := []string{
+		konfluxgen.Truncate(konfluxgen.Sanitize(fmt.Sprintf("%s-%s", "serverless-index", downstreamVersion))),
+	}
+
+	for _, v := range ocpVersions {
+		indexes = append(indexes, konfluxgen.Truncate(konfluxgen.Sanitize(fmt.Sprintf("serverless-index-%s-fbc-%s-index", downstreamVersion, v))))
+	}
```

Comment:
This is just the component name, do you know if it is supposed to work when the application is different?

Reply:
it seems to be working https://github.com/openshift/hac-dev/pull/909

Reply:
I only saw, that the nudge-ref is only a list of strings (without an application ref). As they are in the same namespace, I'd assume so, but not sure.
Let me confirm with them

Reply:
> I only saw, that the nudge-ref is only a list of strings (without an application ref). As they are in the same namespace, I'd assume so, but not sure. Let me confirm with them

Ah didn't see your message https://github.com/openshift-knative/hack/pull/334#discussion_r1799206591

Reply:
from one test I saw
```
 it('should return all components in a namespace', () => {
    useK8sWatchResourceMock.mockReturnValue([mockComponentsData, true, undefined]);
    const { result } = renderHook(() => useAllComponents('test-ns'));
    const [components] = result.current;
    expect(components).toHaveLength(3);
  });
```
so it should work